src/libcore/ptr/const_ptr.rs

```diff
@@ -295,6 +295,64 @@ impl<T: ?Sized> *const T {
         intrinsics::ptr_offset_from(self, origin)
     }
 
+    /// Returns whether two pointers are guaranteed equal. Returns `true` whenever the
+    /// pointers are known to be equal.
+    /// In some contexts (e.g., compile-time evaluation),
+    /// it is not always possible to determine equality of two pointers, so this function may
+    /// spuriously return false for pointers that later actually turn out to be equal.
+    /// But when it returns true, the pointers are guaranteed to be equal.
+    ///
+    /// This function is the mirror of [`guaranteed_ne`], but not its inverse. There are pointer
+    /// comparisons for which both functions return false.
+    ///
+    /// [`guaranteed_ne`]: #method.guaranteed_ne
+    ///
+    /// The return value may change depending on the compiler version and unsafe code may not
+    /// rely on the result of this function for soundness. It is suggested to only use this function
+    /// for performance optimizations where spurious `false` return values by this function do not
+    /// affect the outcome, but just the performance.
+    /// During runtime this function compiles to a direct pointer comparison.
```

Comment:
```suggestion
    /// At runtime this function behaves like `self == other`.
```
But this sounds like it belongs in the very first paragraph, not down here.

Also for full transparency I would add something like this:

> The consequences of using this method to make runtime and compile-time code behave differently have not been explored. This method should *not* be used to introduce such differences, and it should also not be stabilized before we have a better understand of this issue.